### DISCUSSION TITLE
Problem with Sprite & pushimage const convertion

### DISCUSSION THREAD

**Discussion by ghostofperditi0n** [Bodmer/TFT_eSPI#2877] [Q&A - Sprites]
I'm having trouble pushing multiple images to sprites with esp32 on gc9a01 round display. 
In particular, if I use tft.pushimage / push.rotate only with image called "tline" everything works;
Problem appear only on:
background.pushImage(165, 200, 16, 13, set, TFT_WHITE);
background.pushImage(62, 200, 16, 13, folder, TFT_WHITE);
giving:
Compilation error: _invalid conversion from 'const short unsigned int*' to 'uint16_t*' {aka 'short unsigned int* '} [-fpermissive]

Problem disappear drawing on tft using:
tft.pushImage(165, 200, 16, 13, set, TFT_WHITE);
tft.pushImage(62, 200, 16, 13, folder, TFT_WHITE);

In pics.h i upload in flash memory images converted with rinky dink like:

#if defined(__AVR__)
    #include <avr/pgmspace.h>
#elif defined(__PIC32MX__)
    #define PROGMEM
#elif defined(__arm__)
    #define PROGMEM
#endif
const unsigned short thline[3360] PROGMEM={ 0xFFFF, 0xFFFF, 0xFFFF, 0xFFFF, 0xFFFF, 0xFFFF,  ecc ecc ecc ecc }
const unsigned short set[208] PROGMEM={ 0xC618, 0xFFFF, 0xFFFF, ecc ecc ecc ecc }
const unsigned short folder[208] PROGMEM={ 0xC618, 0xC618, 0xC618, 0xC618, ecc ecc ecc ecc }

I paste code:

> #include <SPI.h>
> #include <TFT_eSPI.h>
> #include "pics.h"
> TFT_eSPI tft = TFT_eSPI();
> TFT_eSprite background = TFT_eSprite(&tft);
> TFT_eSprite meter = TFT_eSprite(&tft);
> 
> void setup() {
> 
> tft.begin();
>   background.setColorDepth(8);
>   meter.setColorDepth(8);
>   tft.setRotation(2);
>   meter.createSprite(28,120);
>   background.createSprite(240,240);
>   tft.setSwapBytes(true);
>   meter.setSwapBytes(true);
>   background.setSwapBytes(true);
>  } 
> 
> int thangle=270;
> 
> void loop() {
> 
> uint16_t contorno_color = 50776;
> uint16_t bg_color = 6179;
> background.fillScreen(bg_color);
> 
> 
> 
> meter.pushImage(0, 0, 28, 120, thline);
> tft.setPivot(120,120);
> meter.setPivot(14,120);
> meter.pushRotated(&background, thangle, TFT_WHITE);
> 
>background.pushImage(165, 200, 16, 13, set, TFT_WHITE);
>background.pushImage(62, 200, 16, 13, folder, TFT_WHITE);
> 
> background.pushSprite(0,0);
> delay (5000);
> meter.deleteSprite();
> background.deleteSprite();
> }

Any solutions / ideas ? 
p.s. i was obliged to use :
>   background.setColorDepth(8);
because without i cannot visualize 240 x 240 ( fullscreen )  sprite; 
without limiting bit dept i was able to visualize at max 239 * 239 pixel sprite. ( why ? )
THX

**Comment by ghostofperditi0n** [comment]
background.pushImage(62, 200, 16, 13, folder, TFT_WHITE); ok seem that problem is the trasparency TFT_WHITE. 
Without no problem, with TFT_WHITE, or 65535, or 0xFFFF or 0xFF no way.. invalid conversion error...
Someone can help me please ? thx

**Comment by Bodmer** [comment]
The sprite class does not have a function to push an image with a transparent colour. This should work:

  background.pushImage(165, 200, 16, 13, set);
  background.pushImage(62, 200, 16, 13, folder);

When the sprite is pushed to the screen a 16 bit transparent colour can be specified.
